docs/ui/components/text.tsx

```diff
@@ -0,0 +1,83 @@
+import { css, CSSObject } from '@emotion/react';
+import React, { HTMLAttributes } from 'react';
+
+import { Link, LinkProps } from '~/ui/components/link';
+import { textStyles } from '~/ui/foundations/typography';
+
+export enum TextElements {
+  CODE = 'code',
+  H1 = 'h1',
+  H2 = 'h2',
+  H3 = 'h3',
+  H4 = 'h4',
+  H5 = 'h5',
+  LI = 'li',
+  P = 'p',
+  SPAN = 'span',
+  UL = 'ul',
+  OL = 'ol',
+}
+
+export type TextElementProps = HTMLAttributes<
+  HTMLHeadingElement | HTMLParagraphElement | HTMLLIElement | HTMLUListElement
+> & {
+  testID?: string;
+  size?: 'small' | 'medium' | 'large';
+};
+
+export function createTextComponent(Element: TextElements, textStyle?: CSSObject) {
+  return (props: TextElementProps) => {
+    const { className, testID, size, ...rest } = props;
+
+    let sizeStyles;
+    if (Element === TextElements.P && size) {
+      const sizes = {
+        small: css(textStyles.psmall),
+        medium: css(textStyles.pmedium),
+        large: css(textStyles.plarge),
+      };
+
+      sizeStyles = sizes[size];
+    }
+
+    return (
+      <Element
+        css={[css(textStyles[Element as keyof typeof textStyles]), sizeStyles, css(textStyle)]}
+        className={className}
+        data-testid={testID}
+        {...rest}
+      />
+    );
+  };
+}
+
+// todo(cedric): check with Jon if this is right, we want to use the right semantic element
+export const H1 = createTextComponent(TextElements.H1, textStyles.h2);
```

Comment:
@jonsamp I think this one is important to check. In the design, you picked the styling of a header lower than "semantically correct". We still want to have the right element set, but also properly implement the styling 😄  Is this ok?

Reply:
A pattern we don't use on the website but I've seen elsewhere is to do something like:

```
<H2 as="h1">....</H2>
```

And then the text component will use the `<h1>` element, but use the `h2` styles. We could do the opposite too:

```
<H1 as="h2">...</H1>
```

It would be great to make these semantically correct, thanks for noticing